### PR TITLE
std.json: stringily enum based on tag name

### PR DESCRIPTION
**Pull request by haze**

### CODE REVIEW

**Comment by char** [comment]
i like this

**Comment by Luukdegram** [comment]
Duplicate of https://github.com/ziglang/zig/pull/8661

File: lib/std/json.zig [hunk 1]
```diff
@@ -2797,9 +2797,9 @@ pub fn stringify(
         .Enum => {
             if (comptime std.meta.trait.hasFn("jsonStringify")(T)) {
                 return value.jsonStringify(options, out_stream);
+            } else {
+                return stringify(@tagName(value), options, out_stream);
```
Comment (daurnimator) [inline lib/std/json.zig:2801]: `@tagName` on non-exhaustive enums can be illegal behaviour.

**Review by daurnimator — CHANGES_REQUESTED** [review]
Lets go via the proposal process first (#9451) as noted via the duplicate mentioned, this has been rejected before.

**Pull request closed without merging**